drivers/adc/adc_nrfx_saadc.c

```diff
@@ -494,7 +496,7 @@ static int start_read(const struct device *dev,
 			/* Signal an error if a selected channel has not been
 			 * configured yet.
 			 */
-			if (m_data.positive_inputs[channel_id] == 0U) {
+			if (m_data.positive_inputs[channel_id] == SAADC_INPUT_INVALID) {
```

Comment:
Do we actually need to introduce `SAADC_INPUT_INVALID`? No `NRF_SAADC_*` symbol is defined as 0.

Reply:
Yes, I've noticed the re-purposing of `positive_inputs` and I got the idea, it's good. But my point is that index 0 in `saadc_psels` is not used (`NRF_SAADC_AIN0` is `1`) and there's a check `(channel_cfg->input_positive < NRF_SAADC_AIN0)` performed in `adc_nrfx_channel_setup()`. That's why I don't see the need for `SAADC_INPUT_INVALID`.

Reply:
you are right. I have removed `SAADC_INPUT_INVALID`